### PR TITLE
translations: Write general rules for German.

### PR DESCRIPTION
**Pull request by derAnfaenger**
GCI task: https://codein.withgoogle.com/dashboard/task-instances/5266634199531520/
Zulip discussion: https://chat.zulip.org/#narrow/stream/GCI.20tasks/topic/German.20Translation

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@derAnfaenger, it looks like you've already signed the Dropbox CLA. Thanks!

File: translations/German/general-notes.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+(e.g. *"Du musst auf die Seite ... gehen"* - *"You have to go the page ..."*).
+This keeps the phrases short and avoids unnecessary addressings of the reader.
+
+## Form of titles
+
+**Use continuous titles with verbs in infinitive form**
+
+To be consistent with other online platforms, use continuous titles with verbs
+in infinitive form, e.g. *Manage streams* - *Kanle verwalten* instead of
+~~*Verwalte Kanle*~~.
```
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Should be "Kante" instead of "Kanle" - will fix this after the first review.

File: translations/German/general-notes.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+## Formal or Informal?
+
+**Informal.**
+
+Although written German tends to be quite formal, websites in German are
+usually following informal netiquette. As Zulip's guides are written
+in a more colloquial style, German translations should be rather informal as well.
+
+**Don't use slang or regional phrases in the German translation:**
+* ~~"So'n Dreck kann jedem mal passieren."~~
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I'm not sure we should strike those out, since it's hard to read them - could you rephrase in a more positive way, e.g. "Instead of `<wrong_sentence>` you could say `<right_sentence>`"?

File: translations/German/general-notes.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+* ~~"Das ist die Seite, wo der Quelltext steht."~~
+
+## Form of address
+
+**Use "Du" instead of "Sie".**
+
+For the reasons provided in #Formal_or_Informal, stick to *Du*(informal)
```
Comment (trueskawka) [inline translations/German/general-notes.md]: There should be spaces before brackets everywhere :)

File: translations/German/general-notes.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+instead of *Sie*(formal) when addressing the reader.
+Always capitalize the *Du* when addressing him, do **not** use *du*.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think we could omit the second part of this sentence. 

Also, I usually use gender-neutral "them" and "they" in English when referring to a 3rd person singular of an unspecified gender - that would change to`addressing them`. It makes it easier not to make anyone feel awkward :)

File: translations/German/general-notes.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+## Form of instruction
+
+**Prefer imperative over constructions with auxiliary verbs.**
+
+For instructions, try to use the imperative (e.g. *"Gehe auf die Seite"* -
+*"Go to the page"*) instead of constructions with auxiliary verbs
+(e.g. *"Du musst auf die Seite ... gehen"* - *"You have to go the page ..."*).
+This keeps the phrases short and avoids unnecessary addressings of the reader.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: Also, makes the text more friendly/softer -that might be not the right phrasing, but in general not using auxiliaries is less formal.

File: translations/German/general-notes.md (hunk 1 above)
Comment (trueskawka) [inline translations/German/general-notes.md]: Isn't it `Kanäle`?

File: translations/German/general-notes.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+**Use continuous titles with verbs in infinitive form**
+
+To be consistent with other online platforms, use continuous titles with verbs
+in infinitive form, e.g. *Manage streams* - *Kanle verwalten* instead of
+~~*Verwalte Kanle*~~.
+
+##Concatenation of words
+
+**Try to avoid them.**
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think in this context `concatenation` is singular.

File: translations/German/general-notes.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+The German language is infamous for its concatenations of nouns
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it's famous :) Everyone loves playing around with German.

File: translations/German/general-notes.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+(e.g. "Heizölrückstoßdämpfung", which means "fuel oil recoil attenuation").
```
Comment (trueskawka) [inline translations/German/general-notes.md]: This should be in italics, like the other examples (e.g. line 41).

File: translations/German/general-notes.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+For the sake of correct rendering and simplicity, you should try to avoid such
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think we can emphasize the possible frontend issues more here, maybe give an example of an oddly named button or menu item?

Comment (trueskawka) [inline translations/German/general-notes.md]: Generally German can cause some problems with long phrases on the frontend.

Comment (trueskawka) [inline translations/German/general-notes.md]: Also, I'd say simply `German`, not `The German language`, we're keeping it informal :)

File: translations/German/general-notes.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+## Form of instruction
+
+**Prefer imperative over constructions with auxiliary verbs.**
+
+For instructions, try to use the imperative (e.g. *"Gehe auf die Seite"* -
+*"Go to the page"*) instead of constructions with auxiliary verbs
+(e.g. *"Du musst auf die Seite ... gehen"* - *"You have to go the page ..."*).
+This keeps the phrases short and avoids unnecessary addressings of the reader.
+
+## Form of titles
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think this should be `Form of verbs`, `titles` makes me think of Mr/Mrs.

File: translations/German/general-notes.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+**Try to avoid them.**
+
+The German language is infamous for its concatenations of nouns
+(e.g. "Heizölrückstoßdämpfung", which means "fuel oil recoil attenuation").
+For the sake of correct rendering and simplicity, you should try to avoid such
+concatenations whenever possible.
+
+## Anglicisms
+
+**Use them if a millenial would understand them better than the German equivalent.**
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I'm not sure what your definition of a `millenial` is, but I've heard recently it's those born in the 80's - I don't think you meant that here. A `teenager` might be slightly better, though still a bit baffling for older translators - teenagers tend to use more anglicisms and slang words.

Also, not sure what "understand better" means - e.g. I'd understand "Kanal" and "Stream" just the same, but I have a slight preference for the English word. I know it's hard to define perceived awkwardness. I'd go for "is more widely used in web apps".

This general guide would benefit from the `Task Type A`, which is research for translations - are you planning to take on that task?

File: translations/German/general-notes.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+In contrast to other languages, German happily adapts modern words from other languages,
+especially English. This becomes even more evident in internet applications,
+so you should not be afraid of using them if they provide an advantage over
+the German equivalent. Take the following two examples as a reference:
+
+* Translating *Stream*: Use the German word *Kanal*, since it is just as short
+and perfectly understandable.
+
+* Translating *Bot*: Use *Bot* as an Anglicism, as a completely accurate German
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think `as an Anglicism` is superfluous here.

File: translations/German/general-notes.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+equivalent **doesn't** exist (e.g. *Roboter*) and the term *Bot* is not
+unknown for German speakers.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it's `unknown to`.

File: translations/German/general-notes.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+The German language is infamous for its concatenations of nouns
+(e.g. "Heizölrückstoßdämpfung", which means "fuel oil recoil attenuation").
+For the sake of correct rendering and simplicity, you should try to avoid such
+concatenations whenever possible.
+
+## Anglicisms
+
+**Use them if a millenial would understand them better than the German equivalent.**
+
+In contrast to other languages, German happily adapts modern words from other languages,
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it's more `Unlike other languages` - many languages happily adapt some words from some languages, but not from others, e.g. Muslim-majority countries happily adapt Arabic words, but not English.

File: translations/German/general-notes.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+* Translating *Stream*: Use the German word *Kanal*, since it is just as short
+and perfectly understandable.
+
+* Translating *Bot*: Use *Bot* as an Anglicism, as a completely accurate German
+equivalent **doesn't** exist (e.g. *Roboter*) and the term *Bot* is not
+unknown for German speakers.
+
+## Other
+
+* Try to keep words and phrases short and understandable.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I'd emphasize the frontend issues here again - won't hurt and it's really a pain sometimes.

File: translations/German/general-notes.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+* Be consistent. Use the same terms for the same things, even if that means repeating.
+Have a look into other German translations on Zulip to get a feeling for the vocabulary.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it'd be `have a look at`, though I know why `into` seems to fit better here.

File: translations/German/general-notes.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+* Balance common verbs and nouns with specific IT-related translation of English terms -
```
Comment (trueskawka) [inline translations/German/general-notes.md]: `translation` should be plural - is it a typo in the Polish style guide?

File: translations/German/general-notes.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+this can be tricky, try to check how other resources were translated
+(e.g. GMail, Microsoft websites, Facebook) to decide what wouldnt sound
```
Comment (trueskawka) [inline translations/German/general-notes.md]: `wouldn't`

File: translations/German/general-notes.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+awkward / rude in German.
+
+Some terms are very tricky to translate, so be sure to communicate with other German
+speakers in the community. Its all about making Zulip friendly and usable.
```
Comment (trueskawka) [inline translations/German/general-notes.md]: `It's`

File: translations/German/general-notes.md (hunk 1 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Whoops, yes, you're right. Must have been tired yesterday :P.

File: translations/German/general-notes.md (hunk 2 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Yep, will do so for the other strikings as well.

File: translations/German/general-notes.md (hunk 10 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Yes, `titles` is a little ambiguous. Imo  `form of verbs` doesn't fit a 100% either, since this is referring to the text on buttons, menu-items, etc. What do you think of `Rules for captions`?

File: translations/German/general-notes.md (hunk 11 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Yup, that makes sense. Hm, `teenager` is actually the best alternative here, I think older people will get the meaning of this as well :)

Comment (derAnfaenger) [inline translations/German/general-notes.md]: Oh, okay so I'll just go with the "is more widely used in web apps".

File: translations/German/general-notes.md (hunk 14 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Good point, have never thought about it this way!

File: translations/German/general-notes.md (hunk 17 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Yes, it is, and in the Spanish as well :P (copy-paste culture). I'll make a quick PR to fix them.

File: translations/German/general-notes.md [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+While *ä, ö, ü* and *ß* are more and more being replaced by *ae, oe, ue* and
+*ss* in chats, forums and even websites, German translations who take use of
+the umlauts and *ß* have a more trustworthy appearance. For capitalizations,
+you can replace the *ß* by *ss*.
+
+## False friends
+
+**Watch out!**
+
+False friends for the translation from German to English include
```
Comment (timabbott) [inline translations/German/general-notes.md]: What is a false friend?  I don't understand.

Comment (derAnfaenger) [inline translations/German/general-notes.md]: A false friend is a word in another language that is spelled or sounds similar to a word in one's own language, yet has a different meaning .
Take for example the English word map: it is similar to the german word Mappe, which actually means folder. A map in German is a Landkarte

File: translations/German/general-notes.md (hunk 10 above)
Comment (derAnfaenger) [inline translations/German/general-notes.md]: Or, actually, I think `Form of labels` would be best here.

File: translations/German/general-notes.md [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+
+## Concatenation of words
+
+**Try to avoid it.**
+
+German is famous for its concatenations of nouns
+(e.g. *Heizölrückstoßdämpfung*, which means *fuel oil recoil attenuation*).
+For the sake of correct rendering and simplicity, you should try to avoid such
+concatenations whenever possible, since they can break the layout of the Zulip
+frontend. Try to stick to a maximum length of 20 characters and listen to your
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it's `follow your intuition` in English, though a Polish speaker would listen to it too :)

File: translations/German/general-notes.md [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+intuition.
+
+* A term like *Tastaturkürzel* for *Keyboard shortcuts* is fine - it is shorter than
```
Comment (trueskawka) [inline translations/German/general-notes.md]: This line is > 76 characters.

File: translations/German/general-notes.md [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+20 characters and commonly used in web applications.
+
+* A term like *Benachrichtigungsstichwörter* for *Alert words* should not be used,
+it sounds odd and is longer than 20 characters. You could use "*Stichwörter, die
```
Comment (trueskawka) [inline translations/German/general-notes.md]: Lines 63-64 are too long.

File: translations/German/general-notes.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+* Translating *Bot*: Use *Bot*, as a completely accurate German
+equivalent **doesn't** exist (e.g. *Roboter*) and the term *Bot* is not
+unknown to German speakers.
+
+## Special characters
+
+**Use "ä, ö, ü" and "ß" consistently.**
+
+While *ä, ö, ü* and *ß* are more and more being replaced by *ae, oe, ue* and
+*ss* in chats, forums and even websites, German translations who take use of
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I think it'd be `which make use of` instead of `who take use of`, but I'd simply say `in which the umlauts and  *ß* are used`.

File: translations/German/general-notes.md [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+the umlauts and *ß* have a more trustworthy appearance. For capitalizations,
+you can replace the *ß* by *ss*.
+
+## False friends
+
+**Watch out!**
```
Comment (trueskawka) [inline translations/German/general-notes.md]: I'd explain what false friends are here, you can use the definition from your comment below.

File: translations/German/general-notes.md [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+
+## Other
+
+* Try to keep words and phrases short and understandable. The front-end developers will
+thank you ;)
+
+* Be consistent. Use the same terms for the same things, even if that means repeating.
+Have a look at other German translations on Zulip to get a feeling for the vocabulary.
+
+* Balance common verbs and nouns with specific IT-related translations of English terms -
```
Comment (trueskawka) [inline translations/German/general-notes.md]: This line is long too - I think it's a direct quote from the Polish guide, so it might need reviewing.

**Comment by trueskawka** [comment]
Hi @derAnfaenger!

Thanks for working on this!

Sorry for the delay, as @femnad mentioned in the GCI app there are some possible improvements - could you address those? Thanks!

**Comment by derAnfaenger** [comment]
@trueskawka thanks for your comments, updated the PR :)
I checked the Polish guide, the line-breaks are okay in there, the line was only too long in this German doc.
I also completed the "Research web application translations" task; it basically confirms what this doc says.

**Comment by trueskawka** [comment]
I've seen it, it's very thorough :)

Merging, thanks @derAnfaenger!
